AWS ruby SDK v2 upgrade

dwagner2301 opened the pull request:
API-Calls changed to support the AWS ruby SDK v2. Unsupported config
parameters and exceptions removed.

fixes #11

dwagner2301 committed:
AWS ruby SDK v2 upgrade

API-Calls changed to support the AWS ruby SDK v2. Unsupported config
parameters and exceptions removed.

dwagner2301 commented:
@ph  formal tests are still missing but I'm happy to provide those if somebody points me into the right direction to get it done.

ph commented:
I think the easiest way to make to write the test would be to do an integration test.
What we could do is to use the `logstash-output-sqs` and push events to an SQS queue and get them with the `logstash-input-sqs` and compare them.

I've done something similar for the S3 output at https://github.com/logstash-plugins/logstash-output-s3/blob/master/spec/integration/s3_spec.rb

I make sure all the credentials are in the environment.

What do you think?

ph commented:
I can help tomorrow with the test and show you what I have done.

ph commented on logstash-input-sqs.gemspec:
```diff
@@ -23,7 +23,7 @@ Gem::Specification.new do |s|
 
   s.add_runtime_dependency 'logstash-codec-json'
-  s.add_runtime_dependency 'aws-sdk'
+  s.add_runtime_dependency 'aws-sdk', '~> 2.1.0'
```
We don't need to depend directly on aws-sdk v2.
We can just define a dependencie on logstash-mixin-aws, like this:

``` ruby
  s.add_runtime_dependency "logstash-mixin-aws", ">= 1.0.0"
```

ph commented on lib/logstash/inputs/sqs.rb:
```diff
@@ -137,34 +138,27 @@ def run(output_queue)
 
   def teardown
-    @sqs_queue = nil
+    @sqs = nil
```
I think explicitly doing isn't needed, once the plugin isn't used the GC will clear the memory.

dwagner2301 committed:
Replaces aws-sdk dependencies with mixin-aws

To depend on aws sdk is not necessary. To use logstash-mixin-aws as a
dependency is much better solution.

dwagner2301 committed:
Removes unnecessary de-allocation

The de-allocation is not neccessary since the GC should take care
of it as soon as the plugin isn't used anymore

dwagner2301 commented:
@ph I just added your suggested changes to both plugins. To implement the tests the way you did it with the S3 plugin looks doable. But I won't have a chance to start with that before the weekend. Now, I do understand how to write the tests but I'm still not sure how to run them.

dwagner2301 commented:
@ph I got a dummy test working. I might be able to work on the implementation of the actual integration test by following the S3 example during the next week. Let me know, if you need it done by a specific date and I will try to squeeze it in.

ph commented:
@dwagner2301 If you don't mind, I'll fork your PR and add test to it?

dwagner2301 commented:
@ph I absolutely don't mind! I would have loved to to implement the test but right now it seems you implementing it seems like the best way to get it done as soon as possible.

ph commented:
I am closing this in favor of https://github.com/logstash-plugins/logstash-input-sqs/pull/18
